storage/tests/python/src/norcow.py

```diff
@@ -4,29 +4,55 @@
 from . import consts
 
 
-def align4_int(i: int):
-    return (4 - i) % 4
+def align_int(i: int, align: int):
+    return (align - i) % align
 
 
-def align4_data(data):
-    return data + b"\x00" * align4_int(len(data))
+def align_int_add(i: int, align: int):
+    return i + align_int(i, align)
+
+
+def align_data(data, align: int, padding: bytes = b"\x00"):
+    return data + padding * align_int(len(data), align)
 
 
 class Norcow:
```

Comment:
I think this should be implemented using two derived classes `NorcowBitwise` and `NorcowBytewise` derived from an abstract `Norcow` base class. In the tests the classes can be instantiated using a factory function for example.

Reply:
refactored https://github.com/trezor/trezor-firmware/commit/7ad1c6ae1e81f4975045b862c627881ec1d70de3